Tests/OTelCoreTests/Configuration/OTelConfigurationTests.swift

```diff
@@ -0,0 +1,763 @@
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_SCHEDULE_DELAY": "3000",
+        ]).traces.batchSpanProcessor.scheduleDelay == .seconds(3))
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_SCHEDULE_DELAY": "-3000",
+        ]).traces.batchSpanProcessor.scheduleDelay == OTel.Configuration.default.traces.batchSpanProcessor.scheduleDelay)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_SCHEDULE_DELAY": "0",
+        ]).traces.batchSpanProcessor.scheduleDelay == .zero)
+    }
+
+    // OTEL_BSP_EXPORT_TIMEOUT
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#timeout
+    @Test func testBatchSpanProcessorExportTimeout() {
+        #expect(OTel.Configuration.default.traces.batchSpanProcessor.exportTimeout == .milliseconds(30000))
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_EXPORT_TIMEOUT": "3000",
+        ]).traces.batchSpanProcessor.exportTimeout == .seconds(3))
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_EXPORT_TIMEOUT": "-3000",
+        ]).traces.batchSpanProcessor.exportTimeout == OTel.Configuration.default.traces.batchSpanProcessor.exportTimeout)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_EXPORT_TIMEOUT": "0",
+        ]).traces.batchSpanProcessor.exportTimeout == .zero)
+    }
+
+    // OTEL_BSP_MAX_QUEUE_SIZE
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/
+    // https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#integer
+    @Test func testBatchSpanProcessorMaxQueueSize() {
+        #expect(OTel.Configuration.default.traces.batchSpanProcessor.maxQueueSize == 2048)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_MAX_QUEUE_SIZE": "1024",
+        ]).traces.batchSpanProcessor.maxQueueSize == 1024)
+
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_MAX_QUEUE_SIZE": "0",
+        ]).traces.batchSpanProcessor.maxQueueSize == 0)
+
+        // Negative values should be ignored
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_MAX_QUEUE_SIZE": "-100",
+        ]).traces.batchSpanProcessor.maxQueueSize == OTel.Configuration.default.traces.batchSpanProcessor.maxQueueSize)
+
+        // Invalid values should be ignored
+        #expect(OTel.Configuration.default.applyingEnvironmentOverrides(environment: [
+            "OTEL_BSP_MAX_QUEUE_SIZE": "invalid",
```

Comment:
Do we want to test invalid strings for the previous tests too?

Reply:
Yeah, I got more thorough as I went along. I've gone back and amended the earlier tests.